src/crimson/thread/ThreadPool.h

```diff
@@ -22,26 +22,37 @@ struct WorkItem {
   virtual void process() = 0;
 };
 
-template<typename Func, typename T = std::invoke_result_t<Func>>
+template<typename Func>
 struct Task final : WorkItem {
   Func func;
-  seastar::future_state<T> state;
+  using T = std::invoke_result_t<Func>;
+  using futurator_t = seastar::futurize<T>;
+  typename futurator_t::value_type value;
+  std::exception_ptr e;
```

Comment:
The downside might be a slightly bigger memory footprint for holding `void` as `future_state` uses EBO to knock out `std::tuple<>`. Do we have any performance constraints on the `Task`?

Reply:
fixed and repushed.